aggregating-profiler/src/main/java/ru/fix/aggregating/profiler/engine/CallAggregate.java

```diff
@@ -133,23 +133,7 @@ public void close(AggregatingCall call) {
             return Optional.empty();
         }
 
-        AggregatingCall[] longest = new AggregatingCall[1];
-
-        Set<AggregatingCall> top = new HashSet<>();
-        activeCalls
-                .stream()
-                .sorted(Comparator.comparingLong(AggregatingCall::startNanoTime))
-                .limit(numberOfActiveCallsToTrackAndKeepBetweenReports.get())
-                .forEachOrdered(call -> {
-                    if (top.isEmpty()) {
-                        longest[0] = call;
-                    }
-                    top.add(call);
-                });
-
-        activeCalls.removeIf(call -> !top.contains(call));
-
-        return Optional.ofNullable(longest[0]);
+        return activeCalls.stream().min(Comparator.comparingLong(AggregatingCall::startNanoTime));
```

Comment:
In this calls chain better to remove `reset` part from method names.
Instead of
```
CallAggregate.resetActiveCallsAndGetLongest()  (ru.fix.aggregating.profiler.engine)
CallAggregate.activeCallsMaxLatencyAndResetActiveCalls()  (ru.fix.aggregating.profiler.engine)
CallAggregate.buildReportAndReset(long)  (ru.fix.aggregating.profiler.engine)
```
name methods:
```
CallAggregate.searchLongestActiveCall()  (ru.fix.aggregating.profiler.engine)
CallAggregate.calculateActiveCallsMaxLatency()  (ru.fix.aggregating.profiler.engine)
CallAggregate.buildReportAndReset(long)  (ru.fix.aggregating.profiler.engine)
```

Reply:
refactored